### ISSUE TITLE
Replace raw pointers with smart pointers

### ISSUE THREAD

**Issue by xorz57** [xorz57/forest#8]
There would be no need to make a deconstructor in that way 😃

**Comment by MatusKysel** [comment]
Can you assignee me this issue, it is already fixed I am just waiting till October, thanks :)

**Comment by xorz57** [comment]
Yeah sure! When will you reopen the issue ? :P i cant wait :)

**Comment by MatusKysel** [comment]
Check my last commit, it is fixed

**Comment by xorz57** [comment]
@MatusKysel How do I assign you? I can't choose you as an assignee on the right side bar ... :/

**Comment by MatusKysel** [comment]
Ok, just leave it like that, I will reopen PR on 1st of October, but if you want to try it, it is in my branch

**Comment by xorz57** [comment]
@MatusKysel Let me know if you are still going to reopen a PR.

Don't forget to update the repository !!! (ignore the AVL Tree for the moment. Some things need to be done before smart pointers make it to the AVL Tree). If you would also like to convert the AVL Tree (after it is enchanced) to use smart pointers ... then that's another PR for you 🥂

**Comment by MatusKysel** [comment]
OK, PR is opened and just let me know, when AVL is finalized and I will try to convert it to smart ptrs.

**Comment by xorz57** [comment]
@MatusKysel Sure!!! Whats left to be implemented is to remove the recursion from the avl tree insertion. After that happens, I will let you know :sunglasses:

**Comment by xorz57** [comment]
@MatusKysel Are you ok if I merge NOW this PR now for the 3 data structures you enchanced?
You can open a new PR afterwards for the AVL Tree if you want.

**Comment by MatusKysel** [comment]
Yes, go ahead and merge it :+1:

**Comment by MatusKysel** [comment]
I think you can close this issue now
